Fix SQL in certain locales

plusCubed opened the pull request:
1. I received bug reports of `PRIMARY` being compiled to `prımary` (dotless i), causing errors. Using the US locale for String.toLowerCase() should fix that.  
2. ~~Updated all dependencies (except for bintray release, seems to cause compilation errors with Shadow).~~

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/requery/requery?pullRequest=95) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

npurushe commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
-#Wed Mar 16 20:09:30 PDT 2016
```
This file shouldn't be changed (the gradle distribution is downloaded every ci build should be binary only)

npurushe commented on requery-android/src/main/java/io/requery/android/sqlite/DatabaseSource.java:
```diff
@@ -23,6 +23,11 @@
 import android.database.sqlite.SQLiteOpenHelper;
```
No actual changes to this file

npurushe commented on requery-processor/src/main/java/io/requery/processor/EntityGraph.java:
```diff
@@ -90,7 +91,7 @@ public void add(EntityDescriptor entity) {
             // match by type name
             Optional<TypeKind> primitiveType = Stream.of(TypeKind.values())
                 .filter(TypeKind::isPrimitive)
-                .filter(kind -> kind.toString().toLowerCase().equals(attribute.referencedType()))
```
This can be excluded it can only be primitive type names

npurushe commented on requery/src/main/java/io/requery/sql/QueryBuilder.java:
```diff
@@ -16,12 +16,13 @@
 
 package io.requery.sql;
 
-import io.requery.meta.Attribute;
```
Import order shouldn't change (uses Intellij standard order)

npurushe commented:
Thanks for submitting! I would create 2 PRs as the changing the dependencies is not related to the issue fix, and if the commits are squashed it wouldn't appear as separate commits. Also please see the comments, just a few minor things. Thanks!

plusCubed commented:
@npurushe No problem, fixed and split the dependency updates into pull request #96. Interesting that the Android Studio code style defaults are different from IntelliJ.

npurushe commented:
Looks good thanks!
